File: terceiroSemestre/poo/Exercicio1/ExercicioPython/Ex2.py
from MyMath import sqrt

def baskhara(a, b, delta):
    r1 = (-b+sqrt(delta))/(2*a)
    r2 = (-b-sqrt(delta))/(2*a)
    return r1, r2

class main():

    try:
        a = int(input("Insira a: "))
    except ValueError:
        print("Número inválido!")
        exit()
    try:
        b = int(input("Insira b: "))
    except ValueError:
        print("Número inválido!")
        exit()
    try:
        c = int(input("Insira c: "))
    except ValueError:
        print("Número inválido!")
        exit()
    delta = b*b - (4 * a * c)
    if(delta < 0):
        print("Delta negativo. A função não possui raízes. Insira novos valores. ")
        exit()
    (r1, r2) = baskhara(a, b, delta)
    print("As raízes da equação são {:f} e {:f}.".format(r1, r2))

File: 3rdPeriod/ObjectOrientedProgramming/Exercicio1/src/Ex2.java
import java.io.IOException;

public class Ex2 {
    public static void main(String[] args){
        Double a = 0.0, b = 0.0, c = 0.0;
        try{
            a = EntradaTeclado.leDouble();
        }catch(Exception IOException) {
            System.out.println("Insira um número inteiro!");
        }
        try{
            b = EntradaTeclado.leDouble();
        }catch(Exception IOException) {
            System.out.println("Insira um número inteiro!");
        }
        try{
            c = EntradaTeclado.leDouble();
        }catch(Exception IOException){
            System.out.println("Insira um número inteiro!");
        }

        Double delta = (b*b) - (4*a*c);
        if(delta < 0) {
            System.out.println("Delta negativo");
            return;
        }
        Double[] raizes = MyMath.baskhara(delta, a, b);
        System.out.printf("Raízes são " + raizes[0] + " e " + raizes[1]);
    }



}
